## AIVLE School [AI 미니프로젝트] '서울시 생활정보 기반 대중교통 수요 분석'<br>

---

# 2. 구 별 이동인구 분석
### 서울시 생활이동 정보를 기반으로 서울시 자치구 별 이동인구를 분석합니다

## 데이터 안내

[데이터 파일명]
* 2. seoul_moving.csv

[데이터 정보]
* 서울 시 구별 이동 2022년 4월 데이터

[데이터 컬럼 정보]
* 도착시간 : 0 -> "0:00~0:59"를 의미 
* 출발 자치구 : 자치구 코드
* 성별 : F(여성), M(남성)
* 나이 : 10 ~ 79세 (5세 단위), 0 ~ 10세, 80세 이상으로 구분
* 유형 : H(야간상주지), W(주간상주지), E(기타)
* 평균 이동 시간 : 동일 열 내 이동인구의 월 단위 평균 이동 시간 (10분단위)
* 이동인구 : 추정 합산값으로 소수점 표출, 개인정보 비식별화를 위해 3명 미만(0~3)의 경우 * 처리

---

# 2.1.데이터 불러오기

#### [2.1.1] 데이터 로딩
* Pandas 라이브러리를 활용해서 '2. seoul_moving.csv'파일을 'seoul_moving' 변수에 저장하세요.
    * 데이터 파일 로딩시 참고 사항 
        * 구분자(sep)는 ',' 입니다
        * UTF-8 인코더를 사용해 주세요
        * index column은 0번 열 입니다
    * 데이터를 불러올 때 오류가 발생한다면, 오류에 대해 직접 검색하여 해결해보세요.

In [1]:
import pandas as pd
import numpy as np

In [9]:
# 아래에 코드를 작성하고 결과를 확인합니다.
seoul_moving = pd.read_csv('./2. seoul_moving.csv')

In [8]:
seoul_moving['이동인구(합)'] = seoul_moving['이동인구(합)'].astype('float')

ValueError: could not convert string to float: '*'

In [3]:
# 데이터 프레임의 Shape을 확인합니다.
seoul_moving.shape

(28923054, 11)

In [ ]:
# 아래에 코드를 작성하고 결과를 확인합니다.
# condition = (seoul_moving['도착시간'] > 4)
# only_seoul = only_seoul.loc[condition]
# only_seoul['도착시간'].value_counts()

condition = (seoul_moving['도착시간'] < 5)
df = seoul_moving.loc[condition]
df = df.groupby('요일', as_index=False)['이동인구(합)'].mean()
df

In [ ]:
seoul_moving.info()

In [4]:
condition = (seoul_moving['도착시간'] < 5)
df = seoul_moving.loc[condition]
df.head()

Unnamed: 0    대상연월 요일  도착시간  출발 시군구 코드  도착 시군구 코드 성별  나이 이동유형  평균 이동 시간(분)  \
0           0  202204  일     0      11010      11010  F   0   EE           10   
1           1  202204  일     0      11010      11010  F   0   EH           10   
2           2  202204  일     0      11010      11010  F   0   HE           10   
3           3  202204  일     0      11010      11010  F  10   HW           10   
4           4  202204  일     0      11010      11010  F  10   EH           90   

  이동인구(합)  
0   28.02  
1   27.52  
2   27.83  
3    6.05  
4    5.95

In [5]:
df.groupby('요일', as_index=False)['이동인구(합)'].mean()

C:\Users\user\anaconda3\envs\lecture\lib\site-packages\IPython\core\interactiveshell.py:3457: FutureWarning: Dropping invalid columns in DataFrameGroupBy.mean is deprecated. In a future version, a TypeError will be raised. Before calling .mean, select only columns which should be valid for the function.
  exec(code_obj, self.user_global_ns, self.user_ns)


요일
0  금
1  목
2  수
3  월
4  일
5  토
6  화

---

# 2.2.기본 정보 확인 및 클렌징

+ 데이터 클렌징 : 결측치, 이상치 등을 제거하여 데이터 분석 결과가 왜곡 되는 문제를 방지하기 위한 정제 과정

#### [2.2.1] 기본 정보 확인하기
* 'seoul_moving' 데이터의 정보를 확인해보세요.
* 'describe', 'info', 'head', 'tail' 등 전부 활용해 보세요.

In [4]:
# 아래에 코드를 작성하고 결과를 확인합니다.
seoul_moving.describe()

Unnamed: 0        대상연월          도착시간     출발 시군구 코드     도착 시군구 코드  \
count  2.892305e+07  28923054.0  2.892305e+07  2.892305e+07  2.892305e+07   
mean   7.141390e+05    202204.0  1.365597e+01  1.712362e+04  1.719750e+04   
std    4.753792e+05         0.0  5.849675e+00  9.046478e+03  9.063818e+03   
min    0.000000e+00    202204.0  0.000000e+00  1.101000e+04  1.101000e+04   
25%    3.012810e+05    202204.0  1.000000e+01  1.110000e+04  1.111000e+04   
50%    6.556670e+05    202204.0  1.400000e+01  1.120000e+04  1.120000e+04   
75%    1.082991e+06    202204.0  1.800000e+01  3.101400e+04  3.102100e+04   
max    1.871922e+06    202204.0  2.300000e+01  3.900000e+04  3.900000e+04   

                 나이   평균 이동 시간(분)  
count  2.892305e+07  2.892305e+07  
mean   4.279614e+01  3.279829e+01  
std    1.777485e+01  2.584213e+01  
min    0.000000e+00  1.000000e+01  
25%    3.000000e+01  2.000000e+01  
50%    4.500000e+01  3.000000e+01  
75%    5.500000e+01  4.000000e+01  
max    8.000000e+01  7.700000e+02

In [20]:
# 아래에 코드를 작성하고 결과를 확인합니다.
seoul_moving.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 28923054 entries, 0 to 28923053
Data columns (total 11 columns):
 #   Column       Dtype 
---  ------       ----- 
 0   Unnamed: 0   int64 
 1   대상연월         int64 
 2   요일           object
 3   도착시간         int64 
 4   출발 시군구 코드    int64 
 5   도착 시군구 코드    int64 
 6   성별           object
 7   나이           int64 
 8   이동유형         object
 9   평균 이동 시간(분)  int64 
 10  이동인구(합)      object
dtypes: int64(7), object(4)
memory usage: 2.4+ GB


In [6]:
# 아래에 코드를 작성하고 결과를 확인합니다.
seoul_moving.head()

Unnamed: 0    대상연월 요일  도착시간  출발 시군구 코드  도착 시군구 코드 성별  나이 이동유형  평균 이동 시간(분)  \
0           0  202204  일     0      11010      11010  F   0   EE           10   
1           1  202204  일     0      11010      11010  F   0   EH           10   
2           2  202204  일     0      11010      11010  F   0   HE           10   
3           3  202204  일     0      11010      11010  F  10   HW           10   
4           4  202204  일     0      11010      11010  F  10   EH           90   

  이동인구(합)  
0   28.02  
1   27.52  
2   27.83  
3    6.05  
4    5.95

In [7]:
# 아래에 코드를 작성하고 결과를 확인합니다.
seoul_moving.tail()

Unnamed: 0    대상연월 요일  도착시간  출발 시군구 코드  도착 시군구 코드 성별  나이 이동유형  \
28923049      968458  202204  토    23      39000      11250  M  45   EH   
28923050      968459  202204  토    23      39000      11250  M  50   EH   
28923051      968460  202204  토    23      39000      11250  M  55   EE   
28923052      968461  202204  토    23      39000      11250  M  55   EH   
28923053      968462  202204  토    23      39000      11250  M  75   EH   

          평균 이동 시간(분) 이동인구(합)  
28923049           50    3.16  
28923050           70    3.27  
28923051           50    3.31  
28923052           50    3.28  
28923053           30    5.94

* [참고] 1번 열의 의미
    * " 2022년 4월 일요일에 11010(종로구)에서 11010(종로구)으로 이동하여 0시 ~ 0시 59분 사이에 도착했고, E(기타)에서 H(야간상주지(집))으로 이동한 0 ~ 10세 여성은 총 27.52명이며 이들의 평균 이동시간은 10분이다

#### [2.2.2] 결측치 처리 (NaN이 아닌 결측치 처리)

* '이동인구(합)' column은 이동 인구의 합산인데 숫자가 아닌 object 타입입니다. 

    - 왜 일까요? 

    - Tip! (이동인구 : 추정 합산값으로 소수점 표출, 개인정보 비식별화를 위해 3명 미만 즉, 값이 0~3일 경우 별표 * 처리)

    - 해당 값을 어떻게 처리하는 것이 좋을 지 생각해보아서 각자의 방법으로 변경하세요

    - '인구이동(합)' 컬럼의 데이터 타입을 수치형으로 변경하세요 

In [4]:
# 이동인구(합) 컬럼의 값 별로 개수를 세어봅니다.
seoul_moving['이동인구(합)'].value_counts()

*          3842041
3.3         399374
6.05        366830
3.29        289731
3.31        232069
            ...   
2352.37          1
1617.99          1
3141.86          1
1485.54          1
979.87           1
Name: 이동인구(합), Length: 135812, dtype: int64

In [10]:
# 별표(*)로 표현된 값을 대체합니다.
# 해당 값을 어떻게 처리하는 것이 좋을 지 생각해보아서 각자의 방법으로 변경하세요
#(ex: 최빈값, 평균값도 알 수 없으므로 0~3의 중위값인 1.5로 대체)
seoul_moving.loc[seoul_moving['이동인구(합)'] == '*', '이동인구(합)'] = 1.5

In [10]:
# 이동인구(합) 컬럼의 값 별로 개수를 다시 세어봅니다.
seoul_moving['이동인구(합)'].value_counts()

1.5        3842041
3.3         399374
6.05        366830
3.29        289731
3.31        232069
            ...   
2352.37          1
1617.99          1
3141.86          1
1485.54          1
979.87           1
Name: 이동인구(합), Length: 135812, dtype: int64

In [11]:
# 이제 이동인구(합) column의 데이터 타입을 바꿔주세요
seoul_moving['이동인구(합)'] = seoul_moving['이동인구(합)'].astype('float')
seoul_moving.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 28923054 entries, 0 to 28923053
Data columns (total 11 columns):
 #   Column       Dtype  
---  ------       -----  
 0   Unnamed: 0   int64  
 1   대상연월         int64  
 2   요일           object 
 3   도착시간         int64  
 4   출발 시군구 코드    int64  
 5   도착 시군구 코드    int64  
 6   성별           object 
 7   나이           int64  
 8   이동유형         object 
 9   평균 이동 시간(분)  int64  
 10  이동인구(합)      float64
dtypes: float64(1), int64(7), object(3)
memory usage: 2.4+ GB


In [12]:
condition = (seoul_moving['도착시간'] < 5)
df = seoul_moving.loc[condition]
df = df.groupby('요일', as_index=False)['이동인구(합)'].mean()
df

요일    이동인구(합)
0  금  11.716944
1  목  10.727892
2  수  10.559431
3  월   9.195832
4  일  10.714533
5  토  12.342599
6  화  10.366461

#### [2.2.3] 이동유형, 요일, 성별 변수의 value수 파악

In [12]:
# 이동유형 컬럼의 값 별로 개수를 세어봅니다.
# H(야간상주지), W(주간상주지), E(기타)
seoul_moving['이동유형'].value_counts()

EH    5757073
HE    5520005
EE    5085099
WH    3108026
WE    2809148
HW    2695189
EW    2304535
WW    1026134
HH     617845
Name: 이동유형, dtype: int64

In [13]:
# 요일 컬럼의 값 별로 개수를 세어봅니다.
seoul_moving['요일'].value_counts()

토    4690806
금    4641881
목    4011291
수    3955646
화    3926876
월    3906272
일    3790282
Name: 요일, dtype: int64

* [참고] 2022년 4월의 요일 수는 다음과 같습니다.
    * 일 : 4, 월 : 4, 화 : 4, 수 : 4, 목 : 4, 금 : 5, 토 : 5

In [14]:
# 성별 컬럼의 값 별로 개수를 세어봅니다.
seoul_moving['성별'].value_counts()

M    16355029
F    12568025
Name: 성별, dtype: int64

#### [2.2.4] 심야 이동이 가장 많은 요일 확인

In [7]:
# 아래에 코드를 작성하고 결과를 확인합니다.
# 심야시간은 0:00~ 4:59
condition = (seoul_moving['도착시간'] >= 0) & (seoul_moving['도착시간'] < 5)
seoul_moving.loc[condition, '요일'].value_counts()

토    423995
금    359012
일    331392
목    308651
수    303357
화    292786
월    273062
Name: 요일, dtype: int64

#### [2.2.5] 서울 지역의 데이터만 추출

* 서울시의 데이터만 필요하므로 출발과 도착이 모두 서울인 데이터만 추출하세요.
* 아래 사이트에서 서울시 데이터만 추출하기 위해 필요한 정보를 탐색해보세요. (ex. 자치구 코드 정보)
https://data.seoul.go.kr/dataVisual/seoul/seoulLivingMigration.do

In [8]:
# 출발 시군구 코드, 도착 시군구 코드 활용하여 출발과 도착이 모두 서울인 데이터만 추출해보세요.
condition = (seoul_moving['출발 시군구 코드']//1000 == 11) & (seoul_moving['도착 시군구 코드']//1000 == 11)
only_seoul = seoul_moving.loc[condition]
only_seoul

Unnamed: 0    대상연월 요일  도착시간  출발 시군구 코드  도착 시군구 코드 성별  나이 이동유형  \
0                  0  202204  일     0      11010      11010  F   0   EE   
1                  1  202204  일     0      11010      11010  F   0   EH   
2                  2  202204  일     0      11010      11010  F   0   HE   
3                  3  202204  일     0      11010      11010  F  10   HW   
4                  4  202204  일     0      11010      11010  F  10   EH   
...              ...     ... ..   ...        ...        ... ..  ..  ...   
28875001      920410  202204  토    23      11250      11250  M  80   WH   
28875002      920411  202204  토    23      11250      11250  M  80   EH   
28875003      920412  202204  토    23      11250      11250  M  80   EE   
28875004      920413  202204  토    23      11250      11250  M  80   HE   
28875005      920414  202204  토    23      11250      11250  M  80   WE   

          평균 이동 시간(분)  이동인구(합)  
0                  10    28.02  
1                  10    27.52  
2                  10    27.83  
3                  10     6.05  
4                  90     5.95  
...               ...      ...  
28875001           10    85.10  
28875002           10    28.36  
28875003           20    37.99  
28875004           10     9.42  
28875005           10     9.51  

[10655895 rows x 11 columns]

#### [2.2.6] 서울지역의 구 별로 groupby하여 데이터를 확인해보세요. (유동인구 기준)

* 유출이 제일 많은 구는? - 강남구

* 유입이 제일 많은 구는? - 강남구

* 유출이 제일 적은 구는? - 금천구

* 유입이 제일 적은 구는? - 금천구

* [참고] 서울 구별 코드
    * 11010	종로구
    * 11020	중구
    * 11030	용산구
    * 11040	성동구
    * 11050	광진구
    * 11060	동대문구
    * 11070	중랑구
    * 11080	성북구
    * 11090	강북구
    * 11100	도봉구
    * 11110	노원구
    * 11120	은평구
    * 11130	서대문구
    * 11140	마포구
    * 11150	양천구
    * 11160	강서구
    * 11170	구로구
    * 11180	금천구
    * 11190	영등포구
    * 11200	동작구
    * 11210	관악구
    * 11220	서초구
    * 11230	강남구
    * 11240	송파구
    * 11250	강동구

In [9]:
# 아래에 코드를 작성하고 결과를 확인합니다.
only_seoul.groupby('출발 시군구 코드', as_index=False)['이동인구(합)'].sum().sort_values(
    by='이동인구(합)', ascending=False)

출발 시군구 코드      이동인구(합)
22      11230  46220924.63
23      11240  36207933.02
21      11220  29698071.20
18      11190  25835980.43
13      11140  24517465.86
15      11160  22526485.94
24      11250  22035184.61
10      11110  21392595.19
1       11020  20296645.78
0       11010  19862431.70
20      11210  19706689.55
7       11080  19558088.65
4       11050  18675122.83
3       11040  18121271.09
11      11120  17191013.47
5       11060  16973103.47
12      11130  16952589.75
19      11200  16472620.89
14      11150  15668364.41
2       11030  15148400.31
16      11170  15038178.22
6       11070  14509927.80
8       11090  13084060.28
9       11100  11253816.27
17      11180   8467440.48

In [10]:
# 아래에 코드를 작성하고 결과를 확인합니다.
only_seoul.groupby('도착 시군구 코드', as_index=False)['이동인구(합)'].sum().sort_values(
    by='이동인구(합)', ascending=False)

도착 시군구 코드      이동인구(합)
22      11230  46152613.40
23      11240  36156898.50
21      11220  29763430.80
18      11190  25824111.41
13      11140  24539297.75
15      11160  22770901.22
24      11250  21996357.39
10      11110  21371821.86
1       11020  20275545.43
0       11010  19809988.63
20      11210  19700285.30
7       11080  19524264.96
4       11050  18715186.44
3       11040  18076235.50
11      11120  17168619.43
5       11060  16987577.97
12      11130  16908525.39
19      11200  16460914.02
14      11150  15658707.46
2       11030  15247193.24
16      11170  15011749.82
6       11070  14518503.71
8       11090  13079023.71
9       11100  11243990.02
17      11180   8452662.47

#### [2.2.7] 낮시간만 분석 대상에 넣기 위해 심야버스 제외

* 도착시간 0부터 4까지(0:00 ~ 4:59) 제거

In [11]:
# 아래에 코드를 작성하고 결과를 확인합니다.
condition = (seoul_moving['도착시간'] > 4)
only_seoul = only_seoul.loc[condition]
only_seoul['도착시간'].value_counts()

17    636891
18    630673
16    621016
15    605241
14    597336
19    594995
13    593057
12    567568
11    532959
20    530641
10    505215
21    482814
9     479947
8     444443
22    434670
7     376729
23    366211
6     316162
5     239370
Name: 도착시간, dtype: int64

#### [2.2.8] '도착 시군구 코드' 를 기준으로 각 구별로 '평균 이동 시간(분)' , '이동인구(합)' 의 평균과 합 도출

* groupby 사용, as_index=False로 설정해주세요

In [12]:
# '도착 시군구 코드' 를 기준으로 각 구별로 '평균 이동 시간(분)', '이동인구(합)'의 평균을 도출하여 
# seoul_moving_mean에 저장
seoul_moving_mean = only_seoul.groupby('도착 시군구 코드', as_index=False)[
    ['평균 이동 시간(분)', '이동인구(합)']].mean()
seoul_moving_mean

도착 시군구 코드  평균 이동 시간(분)    이동인구(합)
0       11010    21.349599  44.235978
1       11020    21.012462  44.516072
2       11030    21.364786  36.903883
3       11040    21.740324  42.685596
4       11050    22.474283  46.762548
5       11060    21.673451  39.942708
6       11070    23.412789  41.012753
7       11080    22.160283  45.816273
8       11090    23.811274  39.128470
9       11100    24.489976  37.882283
10      11110    24.029515  55.184590
11      11120    24.611578  50.260428
12      11130    22.026816  42.416672
13      11140    22.044201  54.884906
14      11150    22.793734  46.544142
15      11160    25.124068  58.447282
16      11170    22.933748  41.739382
17      11180    23.998944  30.597875
18      11190    21.615227  56.414634
19      11200    21.867593  40.991815
20      11210    23.377928  49.904036
21      11220    22.831643  64.138624
22      11230    23.094565  90.226636
23      11240    25.429707  83.493581
24      11250    25.670084  64.249687

In [13]:
# '도착 시군구 코드' 를 기준으로 각 구별로 '평균 이동 시간(분)', '이동인구(합)'의 합을 도출하여 
# seoul_moving_sum에 저장
seoul_moving_sum = only_seoul.groupby('도착 시군구 코드', as_index=False)[
    ['평균 이동 시간(분)', '이동인구(합)']].sum()
seoul_moving_sum

도착 시군구 코드  평균 이동 시간(분)      이동인구(합)
0       11010      9356120  19385709.60
1       11020      9306020  19715322.61
2       11030      8514380  14707082.87
3       11040      8887140  17449273.95
4       11050      8625540  17947278.99
5       11060      8838130  16288077.28
6       11070      7939300  13907465.59
7       11080      9072420  18757182.35
8       11090      7649610  12570412.30
9       11100      6995880  10821566.75
10      11110      8975240  20611941.03
11      11120      8077520  16495472.31
12      11130      8465390  16301660.39
13      11140      9494790  23639807.30
14      11150      7379950  15069643.56
15      11160      9388010  21839762.06
16      11170      7946750  14463071.65
17      11180      6362000   8111343.74
18      11190      9581360  25006858.60
19      11200      8408680  15762459.46
20      11210      8772170  18725641.04
21      11220     10291660  28911318.58
22      11230     11437560  44684651.43
23      11240     10610520  34837613.05
24      11250      8430800  21101460.33

In [14]:
print(seoul_moving_mean.columns)
print(seoul_moving_sum.columns)

Index(['도착 시군구 코드', '평균 이동 시간(분)', '이동인구(합)'], dtype='object')
Index(['도착 시군구 코드', '평균 이동 시간(분)', '이동인구(합)'], dtype='object')


In [15]:
# 필요에 따라 column명을 데이터 의미에 맞게 적절히 변경하세요.
seoul_moving_mean.rename(columns={'평균 이동 시간(분)':'구별 이동 시간 평균',
                                  '이동인구(합)':'구별 이동 인구 평균'}, inplace=True)
seoul_moving_sum.rename(columns={'평균 이동 시간(분)':'구별 이동 시간 합',
                                  '이동인구(합)':'구별 이동 인구 합'}, inplace=True)
print(seoul_moving_mean.columns)
print(seoul_moving_sum.columns)

Index(['도착 시군구 코드', '구별 이동 시간 평균', '구별 이동 인구 평균'], dtype='object')
Index(['도착 시군구 코드', '구별 이동 시간 합', '구별 이동 인구 합'], dtype='object')


In [16]:
# seoul_moving_mean, seoul_moving_sum을 합쳐보세요.
df_seoul_moving = pd.merge(seoul_moving_mean, seoul_moving_sum)
df_seoul_moving

도착 시군구 코드  구별 이동 시간 평균  구별 이동 인구 평균  구별 이동 시간 합   구별 이동 인구 합
0       11010    21.349599    44.235978     9356120  19385709.60
1       11020    21.012462    44.516072     9306020  19715322.61
2       11030    21.364786    36.903883     8514380  14707082.87
3       11040    21.740324    42.685596     8887140  17449273.95
4       11050    22.474283    46.762548     8625540  17947278.99
5       11060    21.673451    39.942708     8838130  16288077.28
6       11070    23.412789    41.012753     7939300  13907465.59
7       11080    22.160283    45.816273     9072420  18757182.35
8       11090    23.811274    39.128470     7649610  12570412.30
9       11100    24.489976    37.882283     6995880  10821566.75
10      11110    24.029515    55.184590     8975240  20611941.03
11      11120    24.611578    50.260428     8077520  16495472.31
12      11130    22.026816    42.416672     8465390  16301660.39
13      11140    22.044201    54.884906     9494790  23639807.30
14      11150    22.793734    46.544142     7379950  15069643.56
15      11160    25.124068    58.447282     9388010  21839762.06
16      11170    22.933748    41.739382     7946750  14463071.65
17      11180    23.998944    30.597875     6362000   8111343.74
18      11190    21.615227    56.414634     9581360  25006858.60
19      11200    21.867593    40.991815     8408680  15762459.46
20      11210    23.377928    49.904036     8772170  18725641.04
21      11220    22.831643    64.138624    10291660  28911318.58
22      11230    23.094565    90.226636    11437560  44684651.43
23      11240    25.429707    83.493581    10610520  34837613.05
24      11250    25.670084    64.249687     8430800  21101460.33

In [17]:
# 해당 데이터프레임을 'df_seoul_moving.csv' 파일로 저장하세요
df_seoul_moving.to_csv('df_seoul_moving.csv')

---

# 2.3.데이터 분석하기
+ 데이터의 형태를 살펴보고 다양한 분석기법을 통해 모델링에 적합하도록 정제요소를 선별하세요.
  * 데이터들의 패턴 탐색
  * 변수들간의 관계 파악

In [ ]:
import seaborn as sns
import matplotlib.pyplot as plt

# 시각화 한글폰트 설정
# 맑은 고딕

plt.rc('font', family='Malgun Gothic')
sns.set(font="Malgun Gothic",#"NanumGothicCoding", 
        rc={"axes.unicode_minus":False}, # 마이너스 부호 깨짐 현상 해결
        style='darkgrid')

# scipy
import scipy.stats as spst

#### [2.3.1] 데이터 분포 알아보기
* 다양한 변수를 기준으로 그래프를 그려보고 인사이트를 도출해보세요.

In [ ]:
# 아래에 코드를 작성하고 결과를 확인합니다.


In [ ]:
# 아래에 코드를 작성하고 결과를 확인합니다.

In [ ]:
# 아래에 코드를 작성하고 결과를 확인합니다.

In [ ]:
# 위 차트를 통해 알게된 사실을 정리해봅시다.
# 1. 
# 2.
# 3.

---